src/ir/auth_logic/lowering_ast_datalog.cc

```diff
@@ -262,11 +273,129 @@ std::vector<datalog::Rule> LoweringToDatalogPass::QueriesToDLIR(
   return utils::MapIter<datalog::Rule>(queries, [this](const Query& query) {
     auto [main_pred, not_used] = FactToDLIR(query.principal(), query.fact());
     main_pred = PushPrincipal("says_", query.principal(), main_pred);
-    datalog::Predicate lhs(query.name(), {"dummy_var"}, datalog::kPositive);
+    datalog::Predicate lhs(query.name(), {R"("dummy_var")"},
+                           datalog::kPositive);
     return datalog::Rule(lhs, {main_pred, kDummyPredicate});
   });
 }
 
+std::vector<datalog::RelationDeclaration>
+LoweringToDatalogPass::RelationDeclarationToDLIR(
```

Comment:
This function is too long, can you split this function into multiple smaller ones?  _Roughly_, each loop in this class could it be its own function.